core/arch/arm/include/kernel/generic_boot.h

```diff
@@ -54,9 +54,8 @@ void plat_cpu_reset_late(void);
 void arm_cl2_config(vaddr_t pl310);
 void arm_cl2_enable(vaddr_t pl310);
 
-#ifdef CFG_BOOT_SECONDARY_REQUEST
-extern uint32_t ns_entry_addrs[];
-#endif
+int generic_boot_core_release(size_t core_idx, paddr_t entry);
+paddr_t generic_boot_core_hpen(void);
```

Comment:
if `CFG_BOOT_SECONDARY_REQUEST` is not enable, `generic_boot_core_xxx()` should not be defined. otherwise they should panic but it is not clean to embed code that only aims at panicking.